Tools/cases_generator/generate_cases.py

```diff
@@ -1585,6 +1587,19 @@ def variable_used(node: parser.Node, name: str) -> bool:
     )
 
 
+def no_exception(node: parser.Node) -> bool:
+    if node.name.startswith("JUMP"):
+        return False
+    for token in node.tokens:
+        token_text = token.text.lower()
+        if token.kind == "IDENTIFIER":
+            if token_text == "error_if":
+                return False
+            if token_text.startswith("py") or token_text.startswith("_py"):
```

Comment:
There are static helper functions in `ceval.c` that are used by opcodes and not `Py` prefixed, and some of these can raise. E.g. `match_class`. We can special case the existing ones, but someone could add a new one at any time. So I don't think this heuristic is safe, and I'm not sure I see any possible heuristic that would be safe in the face of possible future changes.

Reply:
I guess the `match_class` case should be covered by `ERROR_IF` used afterward to check if `_PyErr_Occurred`. But what if someone calls a non Py prefixed helper function and then uses a manual `goto error;` rather than `ERROR_IF`? There are some opcodes that use `goto error;` directly...